hera_mc/tests/test_qm.py

```diff
@@ -2,226 +2,213 @@
-                          'test', 5)
-
-        # Test check_metric_desc function to auto-fill descriptions
-        self.test_session.check_metric_desc('test')
-        r = self.test_session.get_metric_desc(metric='test')
-        self.assertEqual(r[0].desc, 'new desc')
-        checkWarnings(self.test_session.check_metric_desc, ['test2'],
-                      message='Metric test2 not found in db')
-        r = self.test_session.get_metric_desc(metric='test2')
-        self.assertTrue('Auto-generated description.' in r[0].desc)
-
-    def test_update_qm_list(self):
-        self.test_session.update_qm_list()
-        r = self.test_session.get_metric_desc()
-        results = []
-        for result in r:
-            self.assertTrue(result.metric in self.metrics_dict)
-            results.append(result.metric)
-        for metric in self.metrics_dict.keys():
-            self.assertTrue(metric in results)
-        metric = list(self.metrics_dict.keys())[0]
-        self.test_session.update_metric_desc(metric, 'foo')
-        self.test_session.commit()
-        # Doing it again will update rather than insert.
-        self.test_session.update_qm_list()
-        r = self.test_session.get_metric_desc(metric=metric)
-        self.assertTrue(r[0].desc == self.metrics_dict[metric])
-
-    def test_ingest_metrics_file(self):
-        # Initialize
-        t1 = Time('2016-01-10 01:15:23', scale='utc')
-        t2 = t1 + TimeDelta(120.0, format='sec')
-        self.obsid = utils.calculate_obsid(t1)
-        # Create obs to satifsy foreign key constraints
-        self.test_session.add_obs(t1, t2, self.obsid)
-        self.test_session.commit()
-        filename = os.path.join(mc.test_data_path, 'example_firstcal_metrics.json')
-        filebase = os.path.basename(filename)
-        self.assertRaises(ValueError, self.test_session.ingest_metrics_file,
-                          filename, 'firstcal')
-        self.test_session.add_lib_file(filebase, self.obsid, t2, 0.1)
-        self.test_session.commit()
-        self.test_session.update_qm_list()
-        self.test_session.ingest_metrics_file(filename, 'firstcal')
-        # Check that things got in
-        firstcal_array_metrics = set(['firstcal_metrics_agg_std_x',
-                                      'firstcal_metrics_good_sol_x',
-                                      'firstcal_metrics_max_std_x'])
-        r = self.test_session.get_array_metric()
-        self.assertEqual(len(r), 3)
-        for result in r:
-            self.assertTrue(result.metric in firstcal_array_metrics)
-        firstcal_ant_metrics = (set(get_firstcal_metrics_dict().keys())
-                                - firstcal_array_metrics)
-        r = self.test_session.get_ant_metric()
-        for result in r:
-            self.assertTrue(result.metric in firstcal_ant_metrics)
-
-
-if __name__ == '__main__':
-    unittest.main()
+metrics_dict = get_metrics_dict()
```

Comment:
Same apprehension with the global definition.

Reply:
fixed.